google-cloud-datastore/src/test/java/com/google/cloud/datastore/DatastoreTest.java

```diff
@@ -648,6 +655,7 @@ private List<RunQueryResponse> buildResponsesForQueryPagination() {
     List<RunQueryResponse> responses = new ArrayList<>();
     RecordQuery<Key> query = Query.newKeyQueryBuilder().build();
     RunQueryRequest.Builder requestPb = RunQueryRequest.newBuilder();
+    requestPb.setProjectId(PROJECT_ID);
```

Comment:
how come this needs to be added?

Reply:
with http transport layer (old way), projectId was being passed to the server in url, `/v1/projects/{projectId}`, there was a way to pass it in the proto payload also but it was not happening earlier.

with grpc transport layer, everything has to be contained in proto payload as url can't be customized. and since we were not passing it in proto payload, emulator rejected the request and this started failing with the below error.

![image](https://github.com/googleapis/java-datastore/assets/5915092/2755f28d-880f-41ad-af7e-fd1de290d369)